src/uu/dd/src/dd.rs

```diff
@@ -0,0 +1,1191 @@
+// This file is part of the uutils coreutils package.
+//
+// (c) Tyler Steele <tyler.steele@protonmail.com>
+//
+// For the full copyright and license information, please view the LICENSE
+// file that was distributed with this source code.
+
+#[macro_use]
+extern crate uucore;
+use uucore::InvalidEncodingHandling;
+
+#[cfg(test)]
+mod dd_unit_tests;
+
+mod datastructures;
+use datastructures::*;
+
+mod parseargs;
+use parseargs::Matches;
+
+mod conversion_tables;
+use conversion_tables::*;
+
+use byte_unit::Byte;
+use clap::{self, crate_version};
+use gcd::Gcd;
+use signal_hook::consts::signal;
```

Comment:
Since this import is unused on platforms other than linux, to get rid of the warnings you have to add `#[cfg(target_os = "linux")]` on it too. This also applies to the other "unused" warnings in this file.

Reply:
Gotcha. This should be taken care of.